cmd/autoowners/main.go

```diff
@@ -196,74 +238,64 @@ func writeYAML(path string, data interface{}, prefix []string) (rerr error) {
-// unmodified.
-func insertStringSlice(insert []string, intoSlice []string,
-	begin int, end int) []string {
-	if begin > end || begin < 0 || end > len(intoSlice) {
-		return intoSlice
-	}
-	firstPart := intoSlice[:begin]
-	secondPart := append(insert, intoSlice[end:]...)
-	return append(firstPart, secondPart...)
-}
-
-// resolveAliases resolves names in the list of owners that
-// match one of the given aliases.  Returns a list of owners
-// with each alias replaced by the list of owners it represents.
-func resolveAliases(aliases *aliases, owners []string) []string {
-	offset := 0 // Keeps track of how many new names we've inserted
-	for i, owner := range owners {
-		if aliasOwners, ok := aliases.Aliases[owner]; ok {
-			index := i + offset
-			owners = insertStringSlice(aliasOwners, owners, index, (index + 1))
-			offset += len(aliasOwners) - 1
+func (orgRepo *orgRepo) resolveOwnerAliases() {
+	if !orgRepo.SimpleConfig.Empty() {
+		orgRepo.SimpleConfig.Approvers = resolveAlias(orgRepo.SimpleConfig.Approvers, orgRepo.RepoAliases)
+		orgRepo.SimpleConfig.Reviewers = resolveAlias(orgRepo.SimpleConfig.Reviewers, orgRepo.RepoAliases)
+		orgRepo.SimpleConfig.RequiredReviewers = resolveAlias(orgRepo.SimpleConfig.RequiredReviewers, orgRepo.RepoAliases)
+	} else {
+		for k, v := range orgRepo.FullConfig.Filters {
+			orgRepo.FullConfig.Filters[k] = repoowners.Config{
+				Approvers:         resolveAlias(v.Approvers, orgRepo.RepoAliases),
+				Reviewers:         resolveAlias(v.Reviewers, orgRepo.RepoAliases),
+				RequiredReviewers: resolveAlias(v.RequiredReviewers, orgRepo.RepoAliases),
+				Labels:            v.Labels,
+			}
 		}
 	}
-	return owners
 }
 
-// resolveOwnerAliases checks whether the orgRepo includes any
-// owner aliases, and attempts to resolve them to the appropriate
-// set of owners.  Returns an owners which replaces any
-// matching aliases with the set of owner names belonging to that alias.
-func (orgRepo *orgRepo) resolveOwnerAliases() *owners {
-	if orgRepo.Aliases == nil || len(orgRepo.Aliases.Aliases) == 0 {
-		return orgRepo.Owners
+func resolveAlias(names []string, alias RepoAliases) []string {
+	if names == nil {
+		return nil
 	}
-
-	return &owners{
-		Approvers:         resolveAliases(orgRepo.Aliases, orgRepo.Owners.Approvers),
-		Reviewers:         resolveAliases(orgRepo.Aliases, orgRepo.Owners.Reviewers),
-		RequiredReviewers: orgRepo.Owners.RequiredReviewers,
-		Labels:            orgRepo.Owners.Labels,
+	result := sets.NewString()
+	for _, name := range names {
+		login := github.NormLogin(name)
+		if val, ok := alias[login]; ok {
+			expand := val.List()
+			logrus.WithField("alias", login).WithField("expand", expand).Debug("resolving alias ...")
+			result.Insert(expand...)
+		} else {
+			result.Insert(login)
+		}
 	}
+	return result.List()
 }
 
 func (orgRepo *orgRepo) writeOwners(whitelist []string) (err error) {
 	for _, directory := range orgRepo.Directories {
 		if inWhitelist(directory, whitelist) {
```

Comment:
nit: this is a blacklist, not a whitelist

Reply:
Done.